src/core/meshcop/meshcop_leader.cpp

```diff
@@ -205,7 +205,8 @@ Error Leader::SetDelayTimerMinimal(uint32_t aDelayTimerMinimal)
 {
     Error error = kErrorNone;
 
-    VerifyOrExit((aDelayTimerMinimal != 0 && aDelayTimerMinimal < DelayTimerTlv::kMinDelay), error = kErrorInvalidArgs);
+    VerifyOrExit((aDelayTimerMinimal != 0 && aDelayTimerMinimal >= DelayTimerTlv::kMinDelay),
```

Comment:
@abtink I will need a ruling on this:
The current code (and associated tests allow/require the argument to be smaller than `DelayTimerTlv::kMinDelay` or thus `OPENTHREAD_CONFIG_TMF_PENDING_DATASET_MINIMUM_DELAY`.
I think that is wrong so the unittest follows my logic whereas the simulation tests still follow the old logic.

Reply:
This seems correct to me as is.

This API is intended for testing only. The idea is to allow setting the `mDelayTimerMinimal` to a value smaller than what is required/specified by the specification (this is the minimum value that will be used and enforced when processing 'pending dataset' requests, ensuring the delay timer is at least this value).

Setting this to smaller values allows tests checking for 'pending dataset' changes to use a shorter delay timer value and run more quickly.

Reply:
then I'll update the unittests accordingly